server/game/cards/01_SOR/units/ReinforcementWalker.ts

```diff
@@ -0,0 +1,48 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { TargetMode, WildcardCardType } from '../../../core/Constants';
+
+
+export default class ReinforcementWalker extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '8691800148',
+            internalName: 'reinforcement-walker',
+        };
+    }
+
+    private deckHasCards(context) {
+        return context.source.controller.drawDeck.length > 0;
+    }
+
+    public override setupCardAbilities() {
+        this.addTriggeredAbility({
+            title: 'Look at the top card of your deck',
+            when: {
+                onCardPlayed: (event, context) => event.card === context.source && this.deckHasCards(context),
```

Comment:
The `deckHasCards` check should be changed, probably to be some form of `thenCondition` inside the `then` ability like in C-3PO. The reason is that the trigger condition here is evaluated some time before the actual resolution of the ability, so it's possible that game state could change in between.

E.g., in this case, there could be some other (hypothetical) triggered ability resolved before this one that puts cards back in the deck, so that by the time this triggered ability would resolve there are cards again and it actually could resolve correctly but we've bypassed it. So our practice is to just have this condition match the bold trigger text as exactly as possible and put conditionals in the body of the ability to halt resolution if certain conditions aren't met.